autonomy/chain/service.py

```diff
@@ -0,0 +1,221 @@
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Helper functions to manage on-chain services"""
+
+from typing import Dict, List, Optional, Tuple
+
+from aea.crypto.base import Crypto, LedgerApi
+from requests.exceptions import ConnectionError as RequestsConnectionError
+from web3.exceptions import ContractLogicError
+
+from autonomy.chain.base import ServiceState, registry_contracts
+from autonomy.chain.config import ChainType, ContractConfigs
+from autonomy.chain.constants import (
+    GNOSIS_SAFE_MULTISIG_CONTRACT,
+    SERVICE_MANAGER_CONTRACT,
+    SERVICE_REGISTRY_CONTRACT,
+)
+from autonomy.chain.exceptions import (
+    InstanceRegistrationFailed,
+    ServiceDeployFailed,
+    ServiceRegistrationFailed,
+)
+from autonomy.chain.mint import transact
+
+
+DEFAULT_DEPLOY_PAYLOAD = "0x"
+
+ServiceInfo = Tuple[int, str, bytes, int, int, int, int, List[int]]
+
+
+def get_agent_instances(
+    ledger_api: LedgerApi, chain_type: ChainType, token_id: int
+) -> Dict:
+    """
+    Get the list of agent instances.
+
+    :param ledger_api: `aea.crypto.LedgerApi` object for interacting with the chain
+    :param chain_type: Chain type
+    :param token_id: Token ID pointing to the on-chain service
+    :returns: number of agent instances and the list of registered addressed
+    """
+
+    return registry_contracts.service_registry.get_agent_instances(
+        ledger_api=ledger_api,
+        contract_address=ContractConfigs.get(SERVICE_REGISTRY_CONTRACT.name).contracts[
+            chain_type
+        ],
+        service_id=token_id,
+    )
+
+
+def get_service_info(
+    ledger_api: LedgerApi, chain_type: ChainType, token_id: int
+) -> ServiceInfo:
+    """
+    Returns service info.
+
+    :param ledger_api: `aea.crypto.LedgerApi` object for interacting with the chain
+    :param chain_type: Chain type
+    :param token_id: Token ID pointing to the on-chain service
+    :returns: security deposit, multisig address, IPFS hash for config,
+            threshold, max number of agent instances, number of agent instances,
+            service state, list of cannonical agents
+    """
+
+    return registry_contracts.service_registry.get_service_information(
+        ledger_api=ledger_api,
+        contract_address=ContractConfigs.get(SERVICE_REGISTRY_CONTRACT.name).contracts[
+            chain_type
+        ],
+        token_id=token_id,
+    )
+
+
+def activate_service(
+    ledger_api: LedgerApi,
+    crypto: Crypto,
+    chain_type: ChainType,
+    service_id: int,
+) -> None:
+    """Activate service."""
```

Comment:
This and the following methods need proper docstrings, including an explanation of what the method does